pkg/scalers/datadog_scaler.go

```diff
@@ -111,6 +115,20 @@ func parseDatadogMetadata(config *ScalerConfig, logger logr.Logger) (*datadogMet
 		return nil, fmt.Errorf("no queryValue given")
 	}
 
+	allowedQueryAggregators := []string{avgString, maxString}
+
+	if val, ok := config.TriggerMetadata["queryAggregator"]; ok && val != "" {
+		queryAggregator := strings.ToLower(val)
```

Comment:
This is a bit weird as go code. I would do just a switch, instead of searching through a slice that just has two elements.

Reply:
It would also remove a full function

Reply:
I had wondered that, and appreciate the comment :)
I'll clean that up as suggested.

Reply:
Side note; it also avoids some code dupe:
```
switch queryAggregator {
    case avgString:
	meta.queryAggegrator = queryAggregator
    case maxString:
	meta.queryAggegrator = queryAggregator
...
```
And it was handy having a slice of accepted values to return the user too (without having a slice of values to return to the user, and then having a matching switch statement for each value too?)